src/task.c

```diff
@@ -610,91 +606,45 @@ task_enqueue(task_system *ts,
 void
 task_perform_all(task_system *ts)
 {
-   if (!ts->use_bg_threads) {
-      platform_status rc;
-      do {
-         rc = task_perform_one(ts);
-         debug_assert(SUCCESS(rc) || STATUS_IS_EQ(rc, STATUS_TIMEDOUT));
-      } while (STATUS_IS_NE(rc, STATUS_TIMEDOUT));
-
-   } else {
-      // wait for bg threads to finish running all the queued up tasks.
-      for (task_type type = TASK_TYPE_FIRST; type != NUM_TASK_TYPES; type++) {
-         task_group *group = &ts->group[type];
-         while (group->current_outstanding_tasks != 0) {
-            platform_sleep(USEC_TO_NSEC(100000)); // 100 msec.
-         }
-      }
-   }
+   platform_status rc;
+   do {
+      rc = task_perform_one(ts);
+      debug_assert(SUCCESS(rc) || STATUS_IS_EQ(rc, STATUS_TIMEDOUT));
+   } while (STATUS_IS_NE(rc, STATUS_TIMEDOUT));
```

Comment:
Add comment explaining that STATUS_TIMEDOUT is being repurposed here to indicate that there are no tasks to run.